### ISSUE TITLE
scylla should check for encryption file paths on start

### ISSUE THREAD

**Issue by ultrabug** [scylladb/scylla#7707]
When scylla starts it should check for the files necessary for encryption to be present when configured.

Instead, it crashes with an obscure error like this:
```
scylla: /jenkins/workspace/scylla-4.2/build/scylla/seastar/include/seastar/core/sharded.hh:493: seastar::sharded<T>::~sharded() [with Service = cql_transport::cql_server]: Assertion `_instances.empty()' failed.
Aborting on shard 0.
Backtrace:
  0x0000000002ef3122
  0x0000000002e976a0
  0x0000000002e97945
  0x0000000002e97990
  0x00007febc5f97a8f
  /opt/scylladb/libreloc/libc.so.6+0x000000000003c9e4
  /opt/scylladb/libreloc/libc.so.6+0x0000000000025894
  /opt/scylladb/libreloc/libc.so.6+0x0000000000025768
  /opt/scylladb/libreloc/libc.so.6+0x0000000000034e75
  0x0000000000d6f47a
  0x0000000000d6f490
  0x0000000000ce6036
  0x0000000000cdf3c3
  0x000000000315bdfc
```

Way to reproduce using docker:

1. create a local scylla configuration directory, modify the scylla.yaml file with:

```
client_encryption_options:
   enabled: true
   certificate: /etc/scylla/scylla_file_does_not_exist.crt
   keyfile: /etc/scylla/scylla.key
```

2. start docker and mount your local scylla folder into the container

```
docker run -v /etc/scylla:/etc/scylla -it scylladb/scylla:4.2.1
```

3. watch a nasty backtrace

**Comment by elcallio** [comment]
Dup of #7211

**Comment by slivne** [comment]
fix is in 4.4 - closing
